Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient.m

```diff
@@ -553,67 +497,46 @@ - (NSURLSessionDataTask *)getCompletionStatusOfCheckoutToken:(NSString *)token c
-			NSInteger statusCode = [(NSHTTPURLResponse *)response statusCode];
-			block(shippingRates, [BUYClient statusForStatusCode:statusCode error:error], error);
-		}];
-	}
-	else {
-		block(nil, BUYStatusUnknown, [NSError errorWithDomain:kShopifyError code:BUYShopifyError_InvalidCheckoutObject userInfo:nil]);
-	}
-	return task;
+	BUYAssertCheckout(checkout);
+	NSURLComponents *components = [self URLComponentsForCheckoutsAppendingPath:@"shipping_rates" checkoutToken:checkout.token queryItems:@{ @"checkout" : @"" }];
+	return [self getRequestForURL:components.URL completionHandler:^(NSDictionary *json, NSURLResponse *response, NSError *error) {
+		NSArray *shippingRates = nil;
+		if (json && !error) {
+			shippingRates = [self.modelManager insertShippingRatesWithJSONArray:json[@"shipping_rates"]];
+		}
+		
+		NSInteger statusCode = [(NSHTTPURLResponse *)response statusCode];
+		block(shippingRates, [BUYClient statusForStatusCode:statusCode error:error], error);
+	}];
 }
 
 #pragma mark - Payments
 
-- (NSURLSessionDataTask *)storeCreditCard:(BUYCreditCard *)creditCard checkout:(BUYCheckout *)checkout completion:(BUYDataCreditCardBlock)block
+- (NSURLSessionDataTask *)storeCreditCard:(BUYCreditCard *)creditCard checkout:(BUYCheckout *)checkout completion:(BUYDataCreditCardBlock)completion
 {
-	NSURLSessionDataTask *task = nil;
+	BUYAssertCheckout(checkout);
+	NSAssert(creditCard, @"Failed to store credit card. No credit card provided.");
 	
-	if ([checkout hasToken] == NO) {
-		block(nil, nil, [NSError errorWithDomain:kShopifyError code:BUYShopifyError_InvalidCheckoutObject userInfo:nil]);
+	NSMutableDictionary *json = [[NSMutableDictionary alloc] init];
+	json[@"token"]            = checkout.token;
+	json[@"credit_card"]      = [creditCard jsonDictionaryForCheckout];
+	if (checkout.billingAddress) {
+		json[@"billing_address"] = [checkout.billingAddress jsonDictionaryForCheckout];
 	}
-	else if (!creditCard) {
-		block(nil, nil, [NSError errorWithDomain:kShopifyError code:BUYShopifyError_NoCreditCardSpecified userInfo:nil]);
-	}
-	else {
-		NSMutableDictionary *json = [[NSMutableDictionary alloc] init];
-		json[@"token"] = checkout.token;
-		json[@"credit_card"] = [creditCard jsonDictionaryForCheckout];
-		if (checkout.billingAddress) {
-			json[@"billing_address"] = [checkout.billingAddress jsonDictionaryForCheckout];
-		}
-		
-		NSError *error = nil;
-		NSData *data = [NSJSONSerialization dataWithJSONObject:@{ @"checkout" : json } options:0 error:&error];
-		if (data && !error) {
-			task = [self postPaymentRequestWithCheckout:checkout body:data completion:block];
-		}
-		else {
-			block(nil, nil, [NSError errorWithDomain:kShopifyError code:BUYShopifyError_InvalidCheckoutObject userInfo:nil]);
-		}
-	}
-	return task;
+	
+	NSData *data = [NSJSONSerialization dataWithJSONObject:@{ @"checkout" : json } options:0 error:nil];
+	
+	NSAssert(data, @"Failed to store credit card. Unable to serialize JSON payload. Possibly invalid credit card object.");
```

Comment:
This assert seems superfluous, since this method controls the creation of the `json` and `data`.